configure-nonroot.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/bash
+
+set -eux
+
+# user and group are from ironic rpms (uid 997, gid 994)
+IRONIC_USER="ironic"
+IRONIC_GROUP="ironic"
+INSPECTOR_GROUP="ironic-inspector"
+
+# certificates (must be writable by both ironic and ironic-inspector)
+mkdir -p /certs
```

Comment:
IIRC this is somewhere that we expect to bind-mount something in to the container. Does creating a directory in the same location and setting the permissions actually have the right/any effect?

Reply:
Thanks for the comment @zaneb !

Bind mounts are actually done for sub-directories under `/certs`, not  `/certs` itself. We are then creating directories in `tls-common.sh` under `/certs` to those bind mount locations, and those mkdir's are failing without these lines.

Two ways of handling this are:

- I can either keep-as is, or 
- Drop the two lines in `configure-nonroot.sh` and also drop L20-24 in https://github.com/metal3-io/ironic-image/blob/ba6890c7b69ef16ed3f362a707a0fdd48a07e6a8/scripts/tls-common.sh#L20 (the failing mkdir's). 

Latter seems to work, but the resulting directory ownerships are bit different. In `configure-nonroot.sh` we assign them to `ironic:ironic-inspector`, but the bind mount created directories and files will be `root:ironic` per the `fsGroup: ...` in the deployment manifest. The sub-directories under `/certs` are only created by the bind mounts, I don't know if that would be an issue in some use-case where mounting does not occur?

I would be more comfortable with the current implementation, but what do you think?

Reply:
Ah, OK, it sounds like we're creating dirs in there so that they exist whether or not somebody actually bind-mounts something at that point. Presumably that is required to not break something else.
I'm comfortable with this implementation. Thanks for the explanation! (I wouldn't be sad if some of this made it's way into a code comment.)

Reply:
I have put much of the explanation into code comments.